config/config_test.go

```diff
@@ -62,17 +78,47 @@ func TestValidConfigs(t *testing.T) {
 func TestInvalidConfigs(t *testing.T) {
 	for k, v := range invalidConfigs {
 		t.Run(k, func(t *testing.T) {
-			filename := mkConfig(t, v)
-			_, err := ReadConfigFile(filename, "", "")
+			filename := mkConfig(t, t.TempDir(), v)
+			_, err := ReadConfigFile(filename, "", "", "")
 			if err == nil {
 				t.Fatal("this config must generate an error")
 			}
 		})
 	}
 }
 
-func mkConfig(t *testing.T, body string) string {
-	f, err := ioutil.TempFile(t.TempDir(), "config_test_*")
+func TestConfigDir(t *testing.T) {
+
+	dir := t.TempDir()
+	f := mkConfig(t, dir, validSubConfig["sub config1"])
+	t.Log(f)
+	f = mkConfig(t, dir, validSubConfig["sub config2"])
+	t.Log(f)
+	emptyConfig := mkConfig(t, dir, "")
+	config, err := ReadConfigFile(emptyConfig, dir, "", "")
+	if err != nil {
+		t.Log("this config must be okay, got error")
+		t.Fail()
+	}
+	// check if key == test exists
+	// check if key == test2 exists
+	foundKey1 := false
+	foundKey2 := false
+	for _, secret := range config.VaultConfig.Secrets {
```

Comment:
Would it be worth doing with a different config type as well to make sure it works when there's say Secrets + AWS in the same file or spread across multiple files.